pkg/snowflake/db.go

```diff
@@ -282,3 +282,79 @@ func (db *DB) RecreateTableOnMaterializationTypeMismatch(ctx context.Context, as
 
 	return nil
 }
+
+func (db *DB) PushColumnDescriptions(ctx context.Context, asset *pipeline.Asset) error {
+	tableComponents := strings.Split(asset.Name, ".")
+	var schemaName string
+	var tableName string
+	switch len(tableComponents) {
+	case 2:
+		schemaName = strings.ToUpper(tableComponents[0])
+		tableName = strings.ToUpper(tableComponents[1])
+	case 3:
+		schemaName = strings.ToUpper(tableComponents[1])
+		tableName = strings.ToUpper(tableComponents[2])
+	default:
+		return nil
+	}
+	anyColumnHasDescription := false
+	colsByName := make(map[string]*pipeline.Column, len(asset.Columns))
+
+	for _, col := range asset.Columns {
+		colsByName[col.Name] = &col
+		if col.Description != "" {
+			anyColumnHasDescription = true
+		}
+	}
+
+	if asset.Description == "" && (len(asset.Columns) == 0 || !anyColumnHasDescription) {
+		return errors.New("no metadata to push: table and columns have no descriptions")
+	}
+
+	queryStr := fmt.Sprintf(
+		`SELECT COLUMN_NAME, COMMENT 
+          FROM %s.INFORMATION_SCHEMA.COLUMNS 
+          WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME = '%s'`,
+		db.config.Database, schemaName, tableName)
+
+	rows, err := db.Select(ctx, &query.Query{Query: queryStr})
+	if err != nil {
+		return errors.Wrapf(err, "failed to query column metadata for %s.%s", schemaName, tableName)
+	}
+
+	existingComments := make(map[string]string)
+	for _, row := range rows {
+		columnName := row[0].(string)
+		comment := ""
+		if row[1] != nil {
+			comment = row[1].(string)
+		}
+		existingComments[columnName] = comment
+	}
+
+	// Find columns that need updates
+	for _, col := range asset.Columns {
+		if col.Description != "" && existingComments[col.Name] != col.Description {
+			updateQuery := fmt.Sprintf(
+				`ALTER TABLE %s.%s.%s MODIFY COLUMN %s COMMENT '%s'`,
+				db.config.Database, schemaName, tableName, col.Name, col.Description,
+			)
+			if err := db.RunQueryWithoutResult(ctx, &query.Query{Query: updateQuery}); err != nil {
+				return errors.Wrapf(err, "failed to update description for column %s", col.Name)
+			}
```

Comment:
maybe we should join multiple queries and run all of them at once? does it work?